include/pybind11/pybind11.h

```diff
@@ -232,6 +229,19 @@ inline std::string generate_function_signature(const char *type_caster_name_fiel
     return signature;
 }
 
+template <typename T>
+inline std::string generate_type_signature() {
+    static constexpr auto caster_name_field = make_caster<T>::name;
+    PYBIND11_DESCR_CONSTEXPR auto descr_types = decltype(caster_name_field)::types();
+
+    const char *text = caster_name_field.text;
```

Comment:
This is used only once: simply inline `caster_name_field.text` in the function all below.

I'd also get rid of all empty lines here.

It would be great if you could add a comment to explain why we're creating a dummy function record.